App/Containers/NewCat.js

```diff
@@ -0,0 +1,309 @@
+import { Alert, DatePickerAndroid, DatePickerIOS, Platform, ScrollView, Switch, Text, TextInput, TouchableHighlight, View } from 'react-native'
+import { Actions as NavigationActions } from 'react-native-router-flux'
+import I18n from 'react-native-i18n'
+import { Colors } from '../Themes'
+import RoundedButton from '../Components/RoundedButton'
+import Dropdown from '../Components/Dropdown'
+import { default as WellCatManager } from '../Services/WellCatManager'
+
+// Styles
+import styles from './Styles/NewCatStyle'
+
+export default class NewCat extends React.Component {
+  constructor () {
+    super()
+    this.state = {
+      name: '',
+      type: 1,
+      breed: -1,
+      gender: -1,
+      dob: new Date(),
+      weight: '',
+      height: '',
+      length: '',
+      declawed: false,
+      outdoor: false,
+      fixed: false,
+      breeds: {},
+      genders: {},
+      timeZoneOffsetInMinutes: (-1) * (new Date().getTimezoneOffset())
+    }
+  }
+
+  componentWillMount () {
+    var that = this
+
+    WellCatManager.getCatBreeds()
+      .then((result) => {
+        if (result.code === 1) {
+          that.setState({
+            breeds: {'items': result.content}
+          })
+        } else {
+          Alert.alert(`${I18n.t('tryLater')}`)
+        }
+      }).catch((error) => {
+        Alert.alert(error)
+      })
+
+    WellCatManager.getAnimalGenders()
+      .then((result) => {
+        if (result.code === 1) {
+          that.setState({
+            genders: {'items': result.content}
+          })
+        } else {
+          Alert.alert(`${I18n.t('tryLater')}`)
+        }
+      }).catch((error) => {
+        Alert.alert(error)
+      })
+  }
+
+  createCat (name, type, breed, gender, dob, weight, height, length, declawed, outdoor, fixed) {
+    var dobFormatted = 'Y-m-d'
+      .replace('Y', dob.getFullYear())
+      .replace('m', dob.getMonth() + 1)
+      .replace('d', dob.getDate())
+
+    WellCatManager.newAnimal(name, type, breed, gender, dobFormatted, weight, height, length, declawed, outdoor, fixed)
+      .then((result) => {
+        if (result.code === 1) {
+          NavigationActions.addCat()
+        } else if (result.code === 0) {
+          Alert.alert(`${I18n.t('unableNewCat')} ${result.content}.`)
+        } else if (result.code === -1) {
+          Alert.alert(`${I18n.t('tryLater')}`)
+        } else {
+          Alert.alert(`${I18n.t('horrible')}`)
+        }
+      })
+  }
+
+  async showPicker () {
+    try {
+      await DatePickerAndroid.open({date: /* this.state.dob */ new Date()}) // commented is undefined
+        .then(({action, year, month, day}) => {
+          if (action !== DatePickerAndroid.dismissedAction) {
+            this.setState({dob: new Date(year, month, day)})  // undefined, date is fine
+          }
+        })
+    } catch (error) {
+      // Alert.alert(`${I18n.t('tryLater')}`)
+      console.log(error)
+    }
+  }
+
+  render () {
+    if (Platform.OS === 'ios') {
```

Comment:
If any of this stuff is common to both ios and android, is it possible to do the conditions inside the render and only have differences where necessary?